### ISSUE TITLE
RESOLVED: npm run dev on Windows 10 / Vagrant 1.9.4 / Homestead 4.0.5

### ISSUE THREAD

**Issue by randelsr** [laravel/docs#3345]
Figure this is going to help out a lot of people trying to figure out why they can't run **npm run dev** or **npm run watch** or **npm run watch-poll** out of the box.

My builds..
Windows 10 Pro 1703.15063.250
VirtualBox: 5.1.20
Vagrant: 1.9.4
Homestead: 4.0.5

Standard errors shown below fix including generic thrown with no recourse. I experienced a ton of strange, generic errors and this resolved them all.

First, be sure to update package.json to reflect the correct cross-env location... my full package.json is shown below the fix as well.

# The Fix

sudo npm install npm@latest -g
curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
sudo apt-get install -y nodejs
rm -rf node_modules 
npm install --no-bin-links **(may need to run twice to overcome max call stack)**
npm rebuild node-sass --no-bin-link
npm run dev

# My package.json:

{
  "private": true,
  "scripts": {
    "dev": "npm run development",
    "development": "node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js",
    "watch": "node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modules/webpack/bin/webpack.js --watch --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js",
    "watch-poll": "npm run watch -- --watch-poll",
    "hot": "node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modules/webpack-dev-server/bin/webpack-dev-server.js --inline --hot --config=node_modules/laravel-mix/setup/webpack.config.js",
    "prod": "npm run production",
    "production": "node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=production node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node_modules/laravel-mix/setup/webpack.config.js"
  },
  "devDependencies": {
    "axios": "^0.15.3",
    "bootstrap-sass": "^3.3.7",
    "cross-env": "^3.2.3",
    "jquery": "^3.1.1",
    "laravel-mix": "0.*",
    "lodash": "^4.17.4",
    "vue": "^2.1.10"
  }
}

# My error messages before the fix:

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs',
1 verbose cli   '/usr/bin/npm',
1 verbose cli   'run',
1 verbose cli   'watch',
1 verbose cli   '--',
1 verbose cli   '--watch-poll' ]
2 info using npm@4.2.0
3 info using node@v7.9.0
4 verbose run-script [ 'prewatch', 'watch', 'postwatch' ]
5 info lifecycle @~prewatch: @
6 silly lifecycle @~prewatch: no script for prewatch, continuing
7 info lifecycle @~watch: @
8 verbose lifecycle @~watch: unsafe-perm in lifecycle true
9 verbose lifecycle @~watch: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/var/www/vhosts/rss3t-server/node_modules$10 verbose lifecycle @~watch: CWD: /var/www/vhosts/rss3t-server
11 silly lifecycle @~watch: Args: [ '-c',
11 silly lifecycle   'node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modules/webpack/bin/w$12 silly lifecycle @~watch: Returned: code: 1  signal: null
13 info lifecycle @~watch: Failed to exec watch script
14 verbose stack Error: @ watch: `node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modules/w$14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:279:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:194:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:194:7)
14 verbose stack     at maybeClose (internal/child_process.js:899:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid @
16 verbose cwd /var/www/vhosts/rss3t-server
17 error Linux 4.4.0-66-generic
18 error argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "watch" "--" "--watch-poll"
19 error node v7.9.0
20 error npm  v4.2.0
21 error code ELIFECYCLE
22 error errno 1
23 error @ watch: `node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modules/webpack/bin/webp$23 error Exit status 1
24 error Failed at the @ watch script 'node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modu$24 error Make sure you have the latest version of node.js and npm installed.
24 error If you do, this is most likely a problem with the  package,
24 error not with npm itself.
24 error Tell the author that this fails on your system:
24 error     node node_modules/cross-env/dist/bin/cross-env.js NODE_ENV=development node_modules/webpack/bin/webpack.js$24 error You can get information on how to open an issue for this project with:
24 error     npm bugs
24 error Or if that isn't available, you can get their info via:
24 error     npm owner ls
24 error There is likely additional logging output above.
25 verbose exit [ 1, true ]
26 warn Local package.json exists, but node_modules missing, did you mean to install?

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'run', 'watch-poll' ]
2 info using npm@4.2.0
3 info using node@v7.9.0
4 verbose run-script [ 'prewatch-poll', 'watch-poll', 'postwatch-poll' ]
5 info lifecycle @~prewatch-poll: @
6 silly lifecycle @~prewatch-poll: no script for prewatch-poll, continuing
7 info lifecycle @~watch-poll: @
8 verbose lifecycle @~watch-poll: unsafe-perm in lifecycle true
9 verbose lifecycle @~watch-poll: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/var/www/vhosts/rss3t-server/node_mo$10 verbose lifecycle @~watch-poll: CWD: /var/www/vhosts/rss3t-server
11 silly lifecycle @~watch-poll: Args: [ '-c', 'npm run watch -- --watch-poll' ]
12 silly lifecycle @~watch-poll: Returned: code: 1  signal: null
13 info lifecycle @~watch-poll: Failed to exec watch-poll script
14 verbose stack Error: @ watch-poll: `npm run watch -- --watch-poll`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:279:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:194:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:194:7)
14 verbose stack     at maybeClose (internal/child_process.js:899:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid @
16 verbose cwd /var/www/vhosts/rss3t-server
17 error Linux 4.4.0-66-generic
18 error argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "watch-poll"
19 error node v7.9.0
20 error npm  v4.2.0
21 error code ELIFECYCLE
22 error errno 1
23 error @ watch-poll: `npm run watch -- --watch-poll`
23 error Exit status 1
24 error Failed at the @ watch-poll script 'npm run watch -- --watch-poll'.
24 error Make sure you have the latest version of node.js and npm installed.
24 error If you do, this is most likely a problem with the  package,
24 error not with npm itself.
24 error Tell the author that this fails on your system:
24 error     npm run watch -- --watch-poll
24 error You can get information on how to open an issue for this project with:
24 error     npm bugs
24 error Or if that isn't available, you can get their info via:
24 error     npm owner ls
24 error There is likely additional logging output above.
25 verbose exit [ 1, true ]
26 warn Local package.json exists, but node_modules missing, did you mean to install?

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'run', 'development' ]
2 info using npm@4.2.0
3 info using node@v7.9.0
4 verbose run-script [ 'predevelopment', 'development', 'postdevelopment' ]
5 info lifecycle @~predevelopment: @
6 silly lifecycle @~predevelopment: no script for predevelopment, continuing
7 info lifecycle @~development: @
8 verbose lifecycle @~development: unsafe-perm in lifecycle true
9 verbose lifecycle @~development: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/var/www/vhosts/rss3t-server/node_m$10 verbose lifecycle @~development: CWD: /var/www/vhosts/rss3t-server
11 silly lifecycle @~development: Args: [ '-c',
11 silly lifecycle   'cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --co$12 info lifecycle @~development: Failed to exec development script
13 verbose stack Error: @ development: `cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress -$13 verbose stack spawn ENOENT
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:33:16)
13 verbose stack     at emitTwo (events.js:106:13)
13 verbose stack     at ChildProcess.emit (events.js:194:7)
13 verbose stack     at maybeClose (internal/child_process.js:899:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
14 verbose pkgid @
15 verbose cwd /var/www/vhosts/rss3t-server
16 error Linux 4.4.0-66-generic
17 error argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "development"
18 error node v7.9.0
19 error npm  v4.2.0
20 error file sh
21 error code ELIFECYCLE
22 error errno ENOENT
23 error syscall spawn
24 error @ development: `cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules -$24 error spawn ENOENT
25 error Failed at the @ development script 'cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progr$25 error Make sure you have the latest version of node.js and npm installed.
25 error If you do, this is most likely a problem with the  package,
25 error not with npm itself.
25 error Tell the author that this fails on your system:
25 error     cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --hide-modules --config=node$25 error You can get information on how to open an issue for this project with:
25 error     npm bugs
25 error Or if that isn't available, you can get their info via:
25 error     npm owner ls
25 error There is likely additional logging output above.
26 verbose exit [ 1, true ]
27 warn Local package.json exists, but node_modules missing, did you mean to install?

0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'run', 'dev' ]
2 info using npm@4.2.0
3 info using node@v7.9.0
4 verbose run-script [ 'predev', 'dev', 'postdev' ]
5 info lifecycle @~predev: @
6 silly lifecycle @~predev: no script for predev, continuing
7 info lifecycle @~dev: @
8 verbose lifecycle @~dev: unsafe-perm in lifecycle true
9 verbose lifecycle @~dev: PATH: /usr/lib/node_modules/npm/bin/node-gyp-bin:/var/www/vhosts/rss3t-server/node_modules/.$10 verbose lifecycle @~dev: CWD: /var/www/vhosts/rss3t-server
11 silly lifecycle @~dev: Args: [ '-c', 'npm run development' ]
12 silly lifecycle @~dev: Returned: code: 1  signal: null
13 info lifecycle @~dev: Failed to exec dev script
14 verbose stack Error: @ dev: `npm run development`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:279:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:194:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:194:7)
14 verbose stack     at maybeClose (internal/child_process.js:899:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
15 verbose pkgid @
16 verbose cwd /var/www/vhosts/rss3t-server
17 error Linux 4.4.0-66-generic
18 error argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "dev"
19 error node v7.9.0
20 error npm  v4.2.0
21 error code ELIFECYCLE
22 error errno 1
23 error @ dev: `npm run development`
23 error Exit status 1
24 error Failed at the @ dev script 'npm run development'.
24 error Make sure you have the latest version of node.js and npm installed.
24 error If you do, this is most likely a problem with the  package,
24 error not with npm itself.
24 error Tell the author that this fails on your system:
24 error     npm run development
24 error You can get information on how to open an issue for this project with:
24 error     npm bugs
24 error Or if that isn't available, you can get their info via:
24 error     npm owner ls
24 error There is likely additional logging output above.
25 verbose exit [ 1, true ]
26 warn Local package.json exists, but node_modules missing, did you mean to install?

Hope this helps somebody...

-Ryan

**Comment by Traxus** [comment]
Hi, thanks for the help. Once I pasted over your package.json and ran the commands my ubuntu install on vagrant atop of windows 10 finally ran npm run dev without error.

It took me 2 days of googling to find this solution, this url really needs some SEO and backlink love.

**Comment by newhwa** [comment]
Amazing!
I was struggling to fix this issue. thanks a lot!!!

**Comment by randelsr** [comment]
Glad it helped you. Not sure why it doesn't get any love from the s. engines but I'm sure it's something many are looking for.

**Comment by Polaris-k** [comment]
aha！
I wasted entire afternoon searching all kinds of solution, only this solve my problem. thanks!!

**Comment by ypeskov** [comment]
Thanks a lot!  In my case as node is installed via nvm was enough just next commands:

sudo npm install npm@latest -g
npm install --no-bin-links (may need to run twice to overcome max call stack)
npm rebuild node-sass --no-bin-link

and then 
npm run dev

And everything works like a charm. I suppose rebuild of saas is the main item here.

Env:
Windows 10
Vagrant 2.0.0 (Ubuntu/xenial)  with mounted source directory of Laravel 5.5

**Comment by ReinierB** [comment]
thanks a lot randelsr, it solved my problem too!

**Comment by nichalko** [comment]
Thanks, sorted!

**Comment by amadeann** [comment]
I kept coming back to this thread for reference for a few months, and then started thinking: why do I even run `npm run x` inside Homestead? 

I think it is not necessary. The output (js & css only) is run in the browser anyway, so there shouldn't be a difference where you compile it. I started compiling js/scss assets directly in PowerShell on Windows and everything works perfectly fine (and faster), without the 'hacks' from the first post (which by the way work really well if for some reason you want to stick with compiling assets in Homestead box).
